tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source_test.ts

```diff
@@ -455,5 +455,33 @@ describe('tb_feature_flag_data_source', () => {
         );
       });
     });
+
+    describe('resetAllPersistedFeatureFlags', () => {
+      it('removes entry from localStorage', () => {
+        const mockStorage: Record<string, string> = {};
+        spyOn(localStorage, 'getItem').and.callFake((key: string) => {
```

Comment:
If James also prefers this test pattern (essentially using a fake instead of a mock) then it could be worth writing a cleanup PR to rewrite the existing tests to use this pattern, too.

Reply:
I would prefer to not have a bunch of test infrastructure which simulates local storage and just test that the expected calls are made.

Reply:
I felt like my version was a bit more thorough, but I've gone ahead and refactored it to make it consistent with the existing tests.

Reply:
FWIW I preferred the use of fakes but +1 for consistency.